dubhash.d

```diff
@@ -0,0 +1,23 @@
+import std.algorithm;
+import std.ascii;
+import std.conv;
+import std.exception;
+import std.file;
+import std.path;
+import std.process;
+import std.range;
+import std.string;
+
+void main()
+{
+    auto dir = environment.get("DUB_PACKAGE_DIR");
```

Comment:
I love this change, but now that this is a separate file, there is a possibility a user runs this file outside of dub. Maybe this could check for dir.length and return if empty?

Reply:
Right now, if the variable doesn't exist, it throws an exception and exits anyway, so it won't try to write a file at some random place in the filesystem (if that's what you want to avoid)

Reply:
if that's the case then this is fine by me, the constant recompilations were getting really annyoing

Reply:
No kidding... dfmt + dscanner + dcd + actual package = 😩